From 53671fd85a33d73699b80564da345ffc9b73c3e5 Mon Sep 17 00:00:00 2001
From: Eli Schwartz <eschwartz@archlinux.org>
Date: Mon, 11 Jun 2018 18:56:50 -0400
Subject: [PATCH] Fix msgpack version constraints using proper setuptools
 metadata.

Embrace environment markers (a.k.a. PEP 508 compliance)! They let you do
sane handling of different requirements for different versions of
python. There's no need to bend over backwards to try to ensure usage of
python34-compatible dependencies on python36.
---
 setup.py | 3 ++-
 1 file changed, 2 insertions(+), 1 deletion(-)

diff --git a/setup.py b/setup.py
index 253a050a..22aabbbe 100644
--- a/setup.py
+++ b/setup.py
@@ -39,7 +39,8 @@
     # we are rather picky about msgpack versions, because a good working msgpack is
     # very important for borg, see https://github.com/borgbackup/borg/issues/3753
     # best versions seem to be 0.4.6, 0.4.7, 0.4.8 and 0.5.6:
-    'msgpack-python >=0.4.6, <=0.5.6, !=0.5.0, !=0.5.1, !=0.5.2, !=0.5.3, !=0.5.4, !=0.5.5',
+    'msgpack-python >=0.4.6, <0.5;python_version=="3.4"',
+    'msgpack >=0.5.6;python_version >="3.5"',
     # if you can't satisfy the above requirement, these are versions that might
     # also work ok, IF you make sure to use the COMPILED version of msgpack-python,
     # NOT the PURE PYTHON fallback implementation: ==0.5.1, ==0.5.4
-- 
2.19.0

